edi_storage/models/edi_backend.py

```diff
@@ -76,3 +81,47 @@ def _component_sort_key(self, component_class):
         return (
             1 if getattr(component_class, "_storage_backend_type", False) else 0,
         ) + res
+
+    def _cron_check_storage_pending_input(self, **kw):
+        for backend in self.filtered(lambda b: b.storage_id):
+            backend._check_storage_pending_input(**kw)
+
+    def _check_storage_pending_input(self, **kw):
+        self.ensure_one()
+        if not self.storage_id or not self.input_dir_pending:
+            _logger.info(
+                "%s ignored. No storage and/or input directory specified.", self.name
+            )
+            return False
+
+        pending_files = self.storage_id.list_files(self.input_dir_pending)
+        exchange_type = self.env["edi.exchange.type"].search(
+            self._domain_exchange_type_storage_pending_input()
+        )
+        if not len(exchange_type) == 1:
```

Comment:
Because of this: https://github.com/OCA/edi/blob/13.0/edi_storage/models/edi_backend.py#L34. It is not currently supported to manage more than one exchange type per edi backend, when that case is covered the logic to find the exchange type could be extended, in the meanwhile I don't see any reason to make it more complex.

Reply:
if the issue is how to find files for a specific type only, we should use the filename pattern to filter the file list from the storage.
I've added this method https://github.com/OCA/storage/blob/13.0/storage_backend/models/storage_backend.py#L114 back in the days having this kind of feature in mind.

Reply:
> Because of this: https://github.com/OCA/edi/blob/13.0/edi_storage/models/edi_backend.py#L34. It is not currently supported to manage more than one exchange type per edi backend, when that case is covered the logic to find the exchange type could be extended, in the meanwhile I don't see any reason to make it more complex.

See my comment above :wink:

Reply:
But when you receive files you cannot assure that the name of the file will be always consistent with a structure defined in Odoo. Imagine you receive bills from your vendors, they will put whatever name to the file (most likely the bill number) and nothing that refers to a exchange type classification pattern. Don't you think?

Reply:
The bill number could have the same format for a given vendor, but it will not be consistent across vendors. Of course I can ask them to do that, but it is better if my software works without needing to bother my suppliers with that and to depend on their development teams to do the task. I prefer to not depend on the vendor's dev to deliver my project to the customer.

Do you think we can support both functionalities, so we cover wider use cases?

Reply:
@LoisRForgeFlow but I would never handle files from different suppliers w/ the same EDI backend. And if you use one backend per source then you can configure different paths per vendor.
Do you see any problem with this?

Reply:
My initial idea is to have one backend shared for several vendors if they use the same data structure (say UBL), and some other backends vendor specific. Now that you highlight it, it is true that I would need a folder per vendor in the shared backends... so seems that your proposal makes more sense.

I have holidays until next week. I will check in detail your proposal using pattern next week.

Thanks for the review and help :wink:

Reply:
Same data structure: this is why we have backend types :wink: 
You can have the same exchange types reusable by backend type.
And you can tie exchange types to specific backend only if you need.

Reply:
>Thanks for the review and help wink

Thanks for your proposal and for accepting my suggestions :wi